src/wrappers/httpwrapper.h

```diff
@@ -0,0 +1,93 @@
+#ifndef ERD_WRAPPER_HTTP_H
+#define ERD_WRAPPER_HTTP_H
+
+#include "http/httplib.h"
+
+#define STATUS_CODE_DEFAULT -1
+#define STATUS_CODE_OK 200
+#define STATUS_MSG_OK "Ok"
+
+namespace wrapper
+{
+namespace http
+{
+
+struct Result
+{
+    int status;
+    bool error;
+    std::string body;
+    std::string statusMessage;
+};
+
+class Client
+{
+public:
+    explicit Client(std::string const &url) : m_client(url.c_str())
+    {}
+
+    Result get(std::string const &path)
+    {
+        int status = STATUS_CODE_DEFAULT;
+        bool err = false;
+        std::string body;
+
+        auto const res = m_client.Get(path.c_str());
+
+        if (res)
+        {
+            status = res->status;
+
+            if (status == STATUS_CODE_OK)
+            {
+                body = res->body;
+            }
+        }
+        else
+        {
+            err = error(res);
+        }
+
+        return Result{status, err, body, getStatusMessage(status)};
+    }
+
+    Result post(std::string const &path, std::string const &message)
+    {
+        int status = STATUS_CODE_DEFAULT;
+        bool err = false;
+        std::string body;
+
+        char const *contentType = "text/plain";
```

Comment:
Not sure if we want this hardcoded - our APIs might accept this `contentType`, but they will always want `application/json` instead.

Do you think we can create an extra wrapper over this wrapper to offer JSON-friendly (our APIs-friendly), generic (with templates) GET & POST, as well? That is, this extra wrapper would also perform some serialization / de-serialization (if the response is successful) or throw exceptions (otherwise)? Or perhaps that is already on the todo-list :)

E.g.
https://github.com/ElrondNetwork/elrond-sdk-erdjs/blob/main/src/proxyProvider.ts
https://github.com/ElrondNetwork/elrond-sdk-erdjs/blob/main/src/apiProvider.ts

Reply:
> Not sure if we want this hardcoded - our APIs might accept this contentType, but they will always want application/json instead.

Indeed, this going to be changed in a future PR and  **POST** method will accept content type as an extra parameter.

> Do you think we can create an extra wrapper over this wrapper to offer JSON-friendly (our APIs-friendly), generic (with templates) GET & POST, as well? That is, this extra wrapper would also perform some serialization / de-serialization (if the response is successful) or throw exceptions (otherwise)? Or perhaps that is already on the todo-list :)

Very good point, while working on the next PR I've noticed this and started working on an extra wrapper for our APIs responses.
However, I will not add it here, since in this PR we just want a wrapper dedicated to a client external library. 
Another wrapper will be created in future PR with the Provider, so it's already on the todo-list.